### DISCUSSION TITLE
Flashing ESP-07: Filesystem does not show up

### DISCUSSION THREAD

**Discussion by besi** [micropython/micropython#14077] [ESP8266] [answered]
I have bought some ESP-07 from Aliexpress. The vendor is "OESAN".

I have some troubles after flashing the device:

## Flashing

1. Enable is connected to 3V3 via 10k pull-up
2. GPIO0 is grounded prior to the RST
3. GPIO15 is shorted to GND
4. RST is briefly shorted to GND 
5. VCC and GND are connected
6. TX and RX are hoocked up to the USB programmer

Now using Thonny I set the parameters as follows
- Install micropython
- Select the Port from the dropdown
- ESP8266
- Version 1.22.2
- (Uncheck) Erase all Flash before... 
- Flash Size: 1MB
- Flash Mode: dout
- Disable Stub loader

After the flashing is done I do get REPL prompt but the filesystem does not show up
and when rebooting I get the following stacktrace:

    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "_boot.py", line 13, in <module>
      File "inisetup.py", line 48, in setup
      File "inisetup.py", line 16, in check_bootsec
      File "flashbdev.py", line 13, in readblocks
    OSError: [Errno 5] EIO

## GENERIC_1MB

Since the chip does have 1MB flash so I tried selecting the `8266-GENERIC_1MB` image from the micropython homepage. But then I don't get a prompt at all upon flashing but the following error message:

> The filesystem starting at sector 153 with size 98 sectors looks corrupt.
You may want to make a flash snapshot and try to recover it. Otherwise,
format it with os.VfsLfs2.mkfs(bdev), or completely erase the flash and
reprogram MicroPython.

## `uos.VfsFat.mkfs`
At no point could I format the filesystem:

    import uos
    import flashbdev
    uos.VfsFat.mkfs(flashbdev.bdev) 

This is also what is happening at the startup when I install the regular image but there is an error.

I also tried the latest preview version `v1.23.0-preview.203.gd712feb68`. Again I get a prompt but no filesystem:

[![enter image description here][1]][1]

  [1]: https://i.stack.imgur.com/fEu81.png

I find it weird that the default image and the 1MB-GENERIC image behave completely differently. Any help regarding this issue would be greatly appreciated.

**Comment by robert-hh** [comment]
If the device has 1MB flash then you have to use the 1MB firmware versions. You may try to use esptool.py instead of Thonny calling it for you , and a simple serial terminal like PicoCom. Then you will get more error messages and will be able to tell the state of the device. Thonny is convenient but of little help if things fail.
Instruction on how to use esptool.py are here: https://docs.micropython.org/en/latest/esp8266/tutorial/intro.html#deploying-the-firmware

**Reply by dpgeorge** [reply, marked as answer]
Yes, you must use the 1MB flash image.

If you get messages about a corrupt filesystem, then use this command to create a new filesystem:
```py
import os
import flashbdev
os.VfsLfs2.mkfs(flashbdev.bdev)
```
Note that it's `VfsLfs2`.

**Reply by besi** [reply]
Thanks very much for taking the time to look at my problem. 
I don't get the chance to issue those commands since if I see the corruption error it's running in a continuous loop without giving me a REPL.
On the other hand if I use the "wrong image" then I do get a REPL but I don't see the filesystem.

**Reply by besi** [reply]
@robert-hh

Thonny does offer various options and passes them on to esptool (as you can see on the screenshot). They have previously improved the install dialog.
<img width="1028" alt="Screenshot 2024-03-13 at 22 13 01" src="https://github.com/micropython/micropython/assets/160412/bed92994-aa11-4859-9565-a34e1b1d2405">

I have flashed devices using esptool before but I don't see which other options or flags that I should use.

The error on reset:
<img width="1072" alt="Screenshot 2024-03-13 at 22 13 56" src="https://github.com/micropython/micropython/assets/160412/0ec38a9e-498c-492b-b868-83bcbf3a7c51">


> sd�l�c|����r�c�c��gn�dog���c8�lrl;lx�o��dcg�|�d�d�b��og��ogd�d`orǛ�'�b�` �g��c�ol���gn�s�ܜ��co�<lćc��g'�d`�ogl n{���gs�lx�o�{�������cg�|ld��c��'o�d`�no�$'{���o��l`r��g��l`����n�{��o|��d�p{lc��|;�l�o��o�d`��;�d�d�$`��s�l��dl sd��sl���b��b|$c��b|��Čllc��o��o'��lo���d�$l���$`�g�b���cl�csd;l{�o����'�߀�#|������:The filesystem starting at sector 153 with size 98 sectors looks corrupt.
You may want to make a flash snapshot and try to recover it. Otherwise,
format it with vfs.VfsLfs2.mkfs(bdev), or completely erase the flash and
reprogram MicroPython.

> The filesystem starting at sector 153 with size 98 sectors looks corrupt.
You may want to make a flash snapshot and try to recover it. Otherwise,
format it with vfs.VfsLfs2.mkfs(bdev), or completely erase the flash and
reprogram MicroPython.

> The filesystem starting at sector 153 with size 98 sectors looks corrupt.
You may want to make a flash snapshot and try to recover it. Otherwise,
format it with vfs.VfsLfs2.mkfs(bdev), or completely erase the flash and
reprogram MicroPython.

> The filesystem starting at sector 153 with size 98 sectors looks corrupt.
You may want to make a flash snapshot and try to recover it. Otherwise,
format it with vfs.VfsLfs2.mkfs(bdev), or completely erase the flash and
reprogram MicroPython.

> The filesystem starting at sector 153 with size 98 sectors looks corrupt.
You may want to make a flash snapshot and try to recover it. Otherwise,
format it with vfs.VfsLfs2.mkfs(bdev), or completely erase the flash and
reprogram MicroPython.

> The filesystem starting at sector 153 with size 98 sectors looks corrupt.
You may want to make a flash snapshot and try to recover it. Otherwise,
format it with vfs.VfsLfs2.mkfs(bdev), or completely erase the flash and
reprogram MicroPython.

> The filesystem starting at sector 153 with size 98 sectors looks corrupt.
You may want to make a flash snapshot and try to recover it. Otherwise,
format it with vfs.VfsLfs2.mkfs(bdev), or completely erase the flash and
reprogram MicroPython.

> The filesystem starting at sector 153 with size 98 sectors looks corrupt.
You may want to make a flash snapshot and try to recover it. Otherwise,
format it with vfs.VfsLfs2.mkfs(bdev), or completely erase the flash and
reprogram MicroPython.

> The filesystem starting at sector 153 with size 98 sectors looks corrupt.
You may want to make a flash snapshot and try to recover it. Otherwise,
format it with vfs.VfsLfs2.mkfs(bdev), or completely erase the flash and
reprogram MicroPython.

> The filesystem starting at sector 153 with size 98 sectors looks corrupt.
You may want to make a flash snapshot and try to recover it. Otherwise,
format it with vfs.VfsLfs2.mkfs(bdev), or completely erase the flash and
reprogram MicroPython.

**Reply by besi** [reply]
I was wrong. I could just `CTRL + C` out of that loop and issue those commands as suggested. Then after a reset everything works as expected!

>     import os
>     import flashbdev
>     os.VfsLfs2.mkfs(flashbdev.bdev)
